src/vs/workbench/contrib/terminal/electron-browser/terminalService.ts

```diff
@@ -117,7 +117,22 @@ export class TerminalService extends BrowserTerminalService implements ITerminal
 		});
 	}
 
-	private _detectWindowsShells(): Promise<IQuickPickItem[]> {
+	/**
+	 * Get the executable file path of shell from registry.
+	 * @param shellName The shell name to get the executable file path
+	 * @returns `[]` or `[ 'path' ]`
+	 */
+	private async _getShellPathFromRegistry(shellName: string): Promise<string[]> {
+		const Registry = await import('vscode-windows-registry');
+
+		try {
+			return [Registry.GetStringRegKey('HKEY_LOCAL_MACHINE', `SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\App Paths\\${shellName}.exe`, '')!];
```

Comment:
While this does compile, you'll still want to put it in a variable and check for undefined just in case as `!` tells the typescript compiler to don't type check this making it less safe. If the key didn't exist (for some reason) this would return [undefined] which could break whatever is consuming the function.

Reply:
Thank you for review,
That was a failure due to my lack of understanding.
I fixed this code.